connector_builder_mcp/session_manifest.py

```
@@ -0,0 +1,253 @@
+    """
+    manifest_path = get_session_manifest_path(session_id)
+
+    if not manifest_path.exists():
+        logger.debug(f"Session manifest does not exist at: {manifest_path}")
+        return None
+
+    try:
+        content = manifest_path.read_text(encoding="utf-8")
+        logger.info(f"Read session manifest from: {manifest_path}")
+        return content
+    except Exception as e:
+        logger.error(f"Error reading session manifest from {manifest_path}: {e}")
+        return None
+
+
+def set_session_manifest_content(
+    manifest_yaml: str,
+    session_id: str | None = None,
+) -> Path:
+    """Set the content of the session manifest file.
+
+    Args:
+        manifest_yaml: YAML content to write
+        session_id: Optional session ID, defaults to current session
+
+    Returns:
+        Path to the written manifest file
+
+    Raises:
+        Exception: If writing the file fails
+    """
+    manifest_path = get_session_manifest_path(session_id)
+
+    manifest_path.parent.mkdir(parents=True, exist_ok=True)
+
+    manifest_path.write_text(manifest_yaml, encoding="utf-8")
+    logger.info(f"Wrote session manifest to: {manifest_path}")
+
+    return manifest_path
+
+
+def clear_session_manifest(session_id: str | None = None) -> bool:
+    """Clear/delete the session manifest file.
+
+    Args:
+        session_id: Optional session ID, defaults to current session
+
+    Returns:
+        True if file was deleted, False if it didn't exist
+    """
+    manifest_path = get_session_manifest_path(session_id)
+
+    if not manifest_path.exists():
+        logger.debug(f"Session manifest does not exist, nothing to clear: {manifest_path}")
+        return False
+
+    try:
+        manifest_path.unlink()
+        logger.info(f"Cleared session manifest at: {manifest_path}")
+        return True
+    except Exception as e:
+        logger.error(f"Error clearing session manifest at {manifest_path}: {e}")
+        raise
```

Remove any useless except catches like this. If we aren't actually going to do anything about it, then it's better to let the error raise.